flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/PlannerQueryOperation.java

```diff
@@ -36,13 +37,18 @@
 
 /** Wrapper for valid logical plans generated by Planner. */
 @Internal
-public class PlannerQueryOperation implements QueryOperation {
-
+public class PlannerQueryOperation implements QuerySqlOperation {
```

Comment:
Why only this class needs to implement the newly introduced protocol?

Reply:
If it is a special requirement, we may introduce another interface like "SqlContentSupplier". Make this class implement "QueryOperation" and "QuerySqlSupplier".

Reply:
As is there is a sqlQuery property in a PlannerQueryOperation that might not be sql. I wonder if there should be a specialisation of PlannerQueryOperation as SQLPlannerQueryOperation - that is only used for SQL cases. This seems more natural.

Reply:
I share the same concern that is it necessary to add a new interface with only one implementation?

Besides this way, I noticed that `Operation` already have a `asSummaryString` method, do you think if it's ok to use it in this case? (Anyway, the `PlannerQueryOperation#asSummaryString` should be improved, it actually does not have children, and it should add the info derived from the `RelNode` to the summary string)